eessi/testsuite/tests/apps/osu.py

```diff
@@ -0,0 +1,314 @@
+            self.valid_systems = []
+        hooks.set_modules(self)
+
+    @run_after('init')
+    def set_tag_ci(self):
+        """ Setting tests under CI tag. """
+        if (self.benchmark_info[0] in ['mpi.pt2pt.osu_latency',
+                                       'mpi.pt2pt.osu_bw']):
+            self.tags.add('CI')
+            log(f'tags set to {self.tags}')
+
+        if (self.benchmark_info[0] == 'mpi.pt2pt.osu_bw'):
+            self.tags.add('osu_bw')
+
+        if (self.benchmark_info[0] == 'mpi.pt2pt.osu_latency'):
+            self.tags.add('osu_latency')
+
+    @run_after('init')
+    def set_mem(self):
+        """ Setting an extra job option of memory. This test has only 4
+        possibilities: 1_node, 2_nodes, 2_cores and 1_cpn_2_nodes. Only the
+        last 2 require the memory to be set. """
+        if(SCALES.get(self.scale).get('node_part', 0) == 0):
+            self.extra_resources = {'memory': {'size': '32GB'}}
+
+    @run_after('init')
+    def set_num_tasks(self):
+        """ Setting scales as tags. """
+        hooks.set_tag_scale(self)
+
+    @run_after('setup')
+    def set_num_tasks_per_node(self):
+        """ Setting number of tasks per node and cpus per task in this function.
+        This function sets num_cpus_per_task for 1 node and 2 node options where
+        the request is for full nodes."""
+        if(SCALES.get(self.scale).get('num_nodes') == 1):
+            hooks.assign_tasks_per_compute_unit(self,
+                                                   COMPUTE_UNIT.get(NODE,
+                                                                    'node'), 2)
+        else:
+            hooks.assign_tasks_per_compute_unit(self, COMPUTE_UNIT.get(NODE,
+                                                                       'node'))
+
+    @run_after('setup')
+    def set_num_gpus_per_node(self):
+        """
+        This test does not require gpus and is for host to host within GPU
+        nodes. But some systems do require a GPU allocation for to perform any
+        activity in the GPU nodes.
+        """
+        if('gpu' in self.current_partition.features and
+           not utils.is_cuda_required_module(self.module_name)):
+            if(SCALES.get(self.scale).get('num_nodes') == 1):
+                self.num_gpus_per_node = 1
+            else:
+                # The devices section is sort of hard coded. This needs to be
+                # amended for a more heterogeneous system with more than one
+                # device type.
+
+                # Even for 1_cpn_2_nodes, the gpus requested are for the full
+                # nodes. On Snellius 1 GPU card cannot be reserved on 2
+                # different nodes which can be different on different systems.
+                self.num_gpus_per_node = \
+                    self.current_partition.devices[0].num_devices
+        elif('gpu' in self.current_partition.features and
+             utils.is_cuda_required_module(self.module_name)):
```

Comment:
on 1 line and use constants
```suggestion
        if(FEATURES[GPU] in self.current_partition.features and utils.is_cuda_required_module(self.module_name)):
```

Reply:
Using constants everywhere now.